src/main/java/com/mraof/minestuck/item/crafting/alchemy/generator/DefaultInterpreter.java

```diff
@@ -33,9 +37,37 @@ public GristSet generateCost(IRecipe<?> recipe, Item output, GenerationContext c
 	{
 		if(recipe.isSpecial())
 			return null;
-		
+
 		GristSet totalCost = new GristSet();
-		for(Ingredient ingredient : recipe.getIngredients())
+
+		// Frick reflection frick reflection frick reflection frick reflection i hate java
+		// this was a whole minecraft oversight anyway
+		// so if they eventually fix this, remove this bit ok?
+		NonNullList<Ingredient> ingredients;
+		if (recipe instanceof SmithingRecipe)
+		{
+			try
+			{
+				Field baseField = SmithingRecipe.class.getDeclaredField("base");
```

Comment:
Because of how minecraft mappings are handled, this won't work in production as the fields won't be called "base" and "addition". You should use `ObfuscationReflectionHelper` instead. (That class will also call `setAccessible()` for you)
You'll need the srg names for that which for "base" is "field_234837_a_" and for "addition" is "field_234838_b_".